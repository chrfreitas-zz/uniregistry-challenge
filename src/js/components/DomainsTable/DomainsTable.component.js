import React, { Component } from 'react';
import PropTypes from 'prop-types';

class DomainsTable extends Component {

    static propTypes = {
        domains: PropTypes.array
    }

    renderHeader() {
        return (
            <th>
                <td>Domains name</td>
                <td>Uniregistry</td>
                <td>Price</td>
            </th>
        )
    }

    renderBody(domains = []) {
        return domains.map((item, index) => {
            return (
                <tr key={index}>
                    <td>item.name</td>
                    <td>item.uniregistry</td>
                    <td>item.price</td>
                </tr>
            );
        });
    }

    render() {
        return (
            <table>
                <thead>
                    { this.renderHeader() }
                </thead>
                <tbody>
                    { this.renderBody(this.props.domains) }
                </tbody>
            </table>
        )
    }
}

export default DomainsTable;