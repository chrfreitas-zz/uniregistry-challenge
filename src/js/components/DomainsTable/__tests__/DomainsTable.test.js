import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import DomainsTable from '../DomainsTable.component';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

describe('DomainsTable component', () => {
    it('It should match with the snapshot', () => {
        const wrapper = renderer.create(<DomainsTable />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('It should render a html list', () => {
        const domains = [
            {
                name: 'foofighter.lol',
                uniregistry: 'Yes',
                price: '$12.00'
            }
        ];

        const wrapper = shallow(<DomainsTable domains={domains} />);
        const amountDomains = domains.length;
        const amountDomainsRendered = wrapper.find('tr').length;

        expect(amountDomains).toEqual(amountDomainsRendered);
    });
});
