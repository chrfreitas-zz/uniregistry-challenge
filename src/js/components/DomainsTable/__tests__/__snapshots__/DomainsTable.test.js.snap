// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainsTable component It should match with the snapshot 1`] = `
<table>
  <thead>
    <th>
      <td>
        Domains name
      </td>
      <td>
        Uniregistry
      </td>
      <td>
        Price
      </td>
    </th>
  </thead>
  <tbody />
</table>
`;
